File: java/muse/util/RandomFactory.java
package muse.util;

public final class RandomFactory {

    private RandomFactory() {}

    public static void seed() {
        // preserve for consistent interface
    }

    public static int genIntN(int min, int max) {
        return min + (int) (Math.random() * (max - min));
    }

    public static int genInt() {
        return genIntN(0, 2_147_483_647);
    }

    public static int genEven() {
        return genInt() >>> 1 << 1;
    }

    public static int genOdd() {
        return genInt() | 1;
    }
}

File: nodejs/muse/util/RandomFactory.js
"use strict"

exports.seed = () => {
    // preserve for consistent interface
}

exports.genIntN = (min, max) => {
    return min + Math.floor(Math.random() * (max - min))
}

exports.genInt = () => {
    return this.genIntN(0, 2147483647)
}

exports.genEven = () => {
    return this.genInt() >>> 1 << 1
}

exports.genOdd = () => {
    return this.genInt() | 1
}
